docs/source-2.0/java/client/configuration.rst

```diff
@@ -0,0 +1,506 @@
+Transports
+----------
+
+Transports manages connections, and handle the sending/receiving of serialized requests/responses.
+
+``ClientTransport``'s can also configure default functionality like adding a user-agent header for HTTP request
+by modifying the client builder using the ``configureClient`` method.
+
+.. admonition:: Important
+    :class: note
+
+    When overriding the configureClient method of a ClientTransport, you need to also call the configureClient
+    method of the ``MessageExchange``, if you want it to take effect. This allows for transports to override
+    or even completely remove ``MessageExchange``-wide functionality.
+
+Transport Discovery
+^^^^^^^^^^^^^^^^^^^
+
+Transport implementations can be automatically discovered by client code generators and dynamic clients via SPI.
+To make a transport implementation discoverable, implement the ``ClientTransportFactory`` service provider.
+
+If no transport is set on a client, the client will attempt to resolve a transport compatible with the current protocol
+from the discoverable transport implementations.
+
+Setting a default transport
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To set a default, add the following to your :ref:`smithy-build.json <smithy-build>`:
+
+.. code-block:: json
+    :caption: smithy-build.json
+
+    "java-client-codegen": {
+        //...
+        "transport": {
+            "http-java": {}
+        }
+    }
+
+.. admonition:: Important
+    :class: note
+
+    Transports set as the default still need to implement the ``ClientTransportFactory`` service provider to
+    be discoverable by the code generation plugin.
+
+Provided transports
+^^^^^^^^^^^^^^^^^^^
+
+* **http-java** - Uses the ``java.net.http.HttpClient`` to send and receive HTTP messages.
+                  This transport is provided by the ``client-http`` module.
+
+
+.. _java-client-authSchemes:
+
+------------
+Auth Schemes
+------------
+
+Auth schemes add authentication/authorization information to a client request. The composition of Auth schemes includes:
+
+1. Scheme ID - A unique identifier for the authentication scheme that should correspond to the ID of a Smithy trait defining an auth scheme (see: https://smithy.io/2.0/spec/authentication-traits.html#smithy-api-authdefinition-trait)
+2. Identity resolver - An API to acquire the customer's identity
+3. Signer - An API to sign requests using the resolved identity.
+
+Auth Schemes can be manually registered on a client at runtime or can be automatically registered by the client code generation plugin. To register an auth scheme at runtime:
+
+.. code-block:: java
+
+    var client = MyClient.builder()
+        .putSupportedAuthSchemes(new MyAuthScheme())
+        .build()
+
+Automatic registration
+^^^^^^^^^^^^^^^^^^^^^^
+
+The Client code generation plugin can discover Auth Schemes on the classpath. If a discovered auth scheme’s ID matches
+an auth scheme ID in the Smithy model it will be automatically registered in the generated client.
```

Comment:
```suggestion
Similar to previous configurations, auth schemes are discovered via SPI. If a discovered auth scheme’s ID matches
an auth scheme ID in the Smithy model, it will be registered in the generated client.
```

Also, 'Auth Scheme' or 'auth scheme', pick one